local/compose/build.go

```diff
@@ -271,10 +272,14 @@ func (s *composeService) toBuildOptions(service types.ServiceConfig, imageTag st
 		plats = append(plats, p)
 	}
 
+	dockerfilePath := service.Build.Dockerfile
+	if filepath.IsAbs(service.Build.Context) && !filepath.IsAbs(dockerfilePath) {
```

Comment:
Any idea why this isn't addressed by https://github.com/compose-spec/compose-go/blob/d07bfddf9d051cee60daf9e6f1b03671d620fb66/loader/normalize.go#L76 ?

Reply:
I ran a quick test.
This code (which should have addressed this during loading) fails as `url.ParseRequestURI` doesn't return error when used with an absolute path!

I'm going to check how buildx distinguish a path vs a git URL

Reply:
seems we will have to replicate https://github.com/docker/buildx/blob/ab09846df70df822fd32b9d0c715dc0605344618/bake/remote.go#L85-L93

Reply:
right, should be... I'll try to debug this (and adapt this PR depending on what I found)

Reply:
proposed a (reasonable?) workaround as https://github.com/compose-spec/compose-go/pull/138 so we won't have to keep compose-go in sync with buildkif supported protocol (which might include some non-documented by the compose-spec)

Reply:
Updated with your fix in compose-go

Reply:
Thx !